src/components/schedule-visualizer/robot.tsx

```diff
@@ -1,84 +1,87 @@
     fill: 'white',
+    /* 1 pixel black shadow to left, top, right and bottom */
+    textShadow: '-1px 0 black, 0 1px black, 1px 0 black, 0 -1px black',
   },
 }));
 
 export interface RobotProps {
   robot: RomiCore.RobotState;
-  footprint: number;
   colorManager: ColorManager;
+  footprint: number;
+  fleetName: string;
   onClick?(e: React.MouseEvent<SVGGElement>, robot: RomiCore.RobotState): void;
 }
 
 const Robot = React.forwardRef(function(
   props: RobotProps,
   ref: React.Ref<SVGGElement>,
 ): React.ReactElement {
+  const resourcesContext = useContext(ResourcesContext);
   const classes = useStyles();
-  const { robot, footprint, colorManager, onClick } = props;
-  const [robotColor, setRobotColor] = React.useState<string | null>(() =>
-    colorManager.robotColorFromCache(robot.name, robot.model),
-  );
+  const { robot, footprint, colorManager, fleetName, onClick } = props;
+  const [iconError, setIconError] = useState(false);
+  const [renderDefaultIcon, setRenderDefaultIcon] = useState(false);
+
+  // The only image formats SVG software support are JPEG, PNG, and other SVG files.
+  const iconPath = useMemo(() => ResourceManager.getRobotIconPath(resourcesContext, fleetName), [
+    resourcesContext,
+    fleetName,
+  ]);
 
-  React.useLayoutEffect(() => {
-    if (robotColor) {
-      return;
-    }
-    (async () => {
-      setRobotColor(await colorManager.robotColor(robot.name, robot.model));
-    })();
-  }, [robot, robotColor, colorManager]);
+  useEffect(() => {
+    setRenderDefaultIcon(!iconPath || !!iconError);
+  }, [iconPath, iconError]);
 
   return (
-    <g
-      ref={ref}
-      data-component="Robot"
-      aria-label={robot.name}
-      onClick={e => onClick && onClick(e, robot)}
-    >
-      {robotColor && (
-        <>
-          <g
-            transform={`translate(${robot.location.x} ${-robot.location.y})
-            rotate(${-(robot.location.yaw * 180) / Math.PI})`}
-          >
-            <filter id={`${robot.name}-shadow`} x="-20%" y="-20%" width="140%" height="140%">
-              <feDropShadow dx="0" dy="0" stdDeviation={footprint * 0.15} floodColor="black" />
-            </filter>
-            <circle
-              className={classes.robotMarker}
-              onClick={e => onClick && onClick(e, robot)}
-              r={footprint}
-              fill={robotColor}
-              filter={`url(#${robot.name}-shadow)`}
-            />
-            <line x2={footprint} stroke="black" strokeWidth="0.05" />
-          </g>
-          <text
-            id="robotName"
-            x={robot.location.x}
-            y={-robot.location.y}
-            className={classes.robotText}
-          >
-            {robot.name.substring(0, 8)}
-          </text>
-        </>
-      )}
-    </g>
+    <>
+      <g
+        ref={ref}
+        data-component="Robot"
+        aria-label={robot.name}
+        onClick={e => onClick && onClick(e, robot)}
+      >
+        {!!iconPath && !iconError && (
```

Comment:
I think this resolves to `!renderDefaultIcon`, using that will make the code more readable.

Reply:
I totally agree with you but I need to check if the `iconPath` exists. If not I got this error. 
```
Type 'string | null' is not assignable to type 'string'.
Type 'null' is not assignable to type 'string'
``` 

Should I use (it's working on both cases):
*  // @ts-ignore on top of `iconPath={iconPath}`
* `!renderDefaultIcon && !!iconPath`. 
* `!!iconPath && !iconError`

Reply:
I see, in that case I'm ok with either option, but I don't like adding ts-ignore, you can just leave this as is.